src/components/encryption.ts

```diff
@@ -196,10 +196,23 @@ export class EncryptedEventBroker {
             // We're not interested in this event, as it's been handled.
             return;
         }
+        this.handleEvent(event.event);
+    }
+
+    private handleEvent(event: WeakEvent) {
         // First come, first serve handling.
-        this.handledEvents.add(key);
-        log.debug(`Handling ${event.getId()} through sync`);
-        this.onEvent(event.event);
+        this.handledEvents.add(`${event.room_id}:${event.event_id}`);
+
+        log.debug(`Handling ${event.event_id} through sync`);
+        this.onEvent(event);
+
+        // Delete the event from the pending list
+        this.eventsPendingSync.delete(event.event_id);
+        let index = this.eventsPendingAS.findIndex((e) => e.event_id === event.event_id);
+        do {
+            this.eventsPendingAS.splice(index, 1);
```

Comment:
Would all of this be possible with `this.eventsPendingAS = this.eventsPendingAS.filter((e) => e.event_id !== event.event_id)`?

Reply:
Yeah, I prefer that.